tests/modules/blockonbackground.c

```diff
@@ -106,7 +132,7 @@ int HelloBlock_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int a
     }
 
     pthread_t tid;
-    RedisModuleBlockedClient *bc = RedisModule_BlockClient(ctx,HelloBlock_Reply,HelloBlock_Timeout,HelloBlock_FreeData,timeout);
+    RedisModuleBlockedClient *bc = RedisModule_BlockClient(ctx,NULL,NULL,HelloBlock_FreeData,0);
```

Comment:
now timeout and reply are all controlled by the thread, so we don't need these two callbacks and timeout.

Reply:
i think the point of that test was to interact with Redis's timeout mechanism.

Reply:
Yes, that's exactly what I was trying to say, it wants to reply to the user in the middle of sleep of module thread.
If we want to avoid the main thread triggering a timeout and let the module thread handle the time measure itself, we have to drop the timeout callback.
That's what I'm struggling with.

Reply:
@oranagra To ensure the point of that test, I think we need to add lock to protect it.
First, we should clarify that RM_BlockedClientMeasureTime*() API are not thread-safe.
Since module developers know they will use these APIs in multiple threads, they should ensure these APIs are safe to use.
Frequently locking and unlocking the GIL is a bad idea, but using a local lock owned by the module might not be a bad idea.

Reply:
ok, i'm fine with that.
please add the comment in the API docs

Reply:
done.